dropshot/src/api_description.rs

```diff
@@ -956,17 +961,17 @@ fn j2oas_object(
  * document may be generated via [`write()`](`OpenApiDefinition::write`) or
  * [`json()`](`OpenApiDefinition::json`).
  */
-pub struct OpenApiDefinition<'a> {
-    api: &'a ApiDescription,
+pub struct OpenApiDefinition<'a, Context: ServerContext> {
```

Comment:
I *think* this is possible by breaking the `OpenApiDefinition`, and all related "needed traits" into a separate non-context-dependent trait.

However, this is currently really difficult, because `OpenApiDefinition` relies on a very concrete `HttpRouter` to do this work.

Unless folks strongly oppose, I agree, this would be better suited to a follow-up change (as it will require decoupling the notion of an "abstract endpoint" - no context, used w/openapi - from a "concrete" one)